tutorials/chapter1.md

```diff
@@ -89,38 +98,52 @@ This examples shows several features of Klepsydra already. Let's have a look at
-      }
-   );
-   
-   eventloop.start();
-   
-   SimplePublisher simplePublisher(eventloop.getPublisher<std::string>("example2", 0, nullptr, nullptr));
-
-   simplePublisher.run();
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> eventloop(nullptr);
 
-   eventloop.getSubscriber<std::string>()->removeListener("example1");
+    std::cout << "Main thread ID: " << std::this_thread::get_id() << std::endl;
 
-   eventloop.stop();
+    eventloop.start();
+    eventloop.getSubscriber<std::string>("example2")->registerListener("listener",
+                                                             [](const std::string & message) {
+                                                                 std::cout << "Message received: " << message << std::endl;
+    								 std::cout << "Eventloop (subscriber) thread ID: " << std::this_thread::get_id() << std::endl;
+                                                             });
+    std::this_thread::sleep_for(std::chrono::milliseconds(1)); // Ensures listener has been registered before publisher runs.
+
+    SimplePublisher simplePublisher(eventloop.getPublisher<std::string>("example2", 0, nullptr, nullptr));
+    simplePublisher.run();
+    std::this_thread::sleep_for(std::chrono::milliseconds(1)); // Ensures published message ALWAYS reaches subscriber before the listener is removed in next line.
+
+    eventloop.getSubscriber<std::string>("example2")->removeListener("listener");
+    eventloop.stop();
 }
 ```
 
 This version of the main is slighlty different from the previous one in the following elements:
 
 * The instatiation of the ```EventLoopMiddlewareProvider``` does not take any template, nor the ```std::function``` pair for the smart object pool. The reason for that is that the ```EventLoop``` is a general memory sharing system that can handle data of any type, it only requires the size of the ring buffer as template parameter. It is only when invoking ```getPublisher``` and ```getSubscriber``` that types need to be specified. The ```EventEmitter``` can only handle one publisher / subcsriber pair, this is why the templating and smart pool configuration happen at the declaration and construction level.
-* Another important difference is that the event loop requires starting and stopping. The start will create and start the listening thread, which is the only thread that will handle all the listeners associated to all subscribers. It is important that, before stopping, all listeners have been removed.
+* Another important difference is that the event loop requires starting and stopping. The start will create and start the listening thread, which is the only thread that will handle all the listeners
+
+In this case, the number of threads increase. There are two threads running:
+
+* The main thread
+* The listener thread
+
+
+ associated to all subscribers. It is important that, before stopping, all listeners have been removed.
```

Comment:
the paragraph was cut in the middle, as it seems....

Reply:
Done